tests/test_decision_making/test_utility_functions/test_utility_functions.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2023 The GPJax Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from jax import config
+
+config.update("jax_enable_x64", True)
+
+from beartype.typing import Type
+import jax.random as jr
+import pytest
+
+from gpjax.decision_making.test_functions.continuous_functions import (
+    AbstractContinuousTestFunction,
+    Forrester,
+    LogarithmicGoldsteinPrice,
+)
+from gpjax.decision_making.utility_functions.base import (
+    AbstractSinglePointUtilityFunctionBuilder,
+)
+from gpjax.decision_making.utility_functions.probability_of_improvement import (
+    ProbabilityOfImprovement,
+)
+from gpjax.decision_making.utility_functions.thompson_sampling import ThompsonSampling
+from gpjax.decision_making.utils import OBJECTIVE
+from gpjax.typing import KeyArray
+from tests.test_decision_making.utils import (
+    generate_dummy_conjugate_posterior,
+    generate_dummy_non_conjugate_posterior,
+)
+
+
+@pytest.mark.filterwarnings(
```

Comment:
Nice, this will help make testing additional utility functions a lot easier.

Reply:
I'm glad!